File: frontend/src/app/shared/models/rolls/attack-damage-roll-request.ts
import {RollRequest} from './roll-request';

export class AttackDamageRollRequest {
  attack: RollRequest;
  damage: RollRequest;

  constructor(attack: RollRequest, damage: RollRequest) {
    this.attack = attack;
    this.damage = damage;
  }
}

File: backend/src/main/java/com/herd/squire/models/rolls/AttackDamageRollRequest.java
package com.herd.squire.models.rolls;

public class AttackDamageRollRequest {
    private RollRequest attack;
    private RollRequest damage;

    public AttackDamageRollRequest() {}

    public AttackDamageRollRequest(RollRequest attack, RollRequest damage) {
        this.attack = attack;
        this.damage = damage;
    }

    public RollRequest getAttack() {
        return attack;
    }

    public void setAttack(RollRequest attack) {
        this.attack = attack;
    }

    public RollRequest getDamage() {
        return damage;
    }

    public void setDamage(RollRequest damage) {
        this.damage = damage;
    }
}
